ethcore/src/state.rs

```diff
@@ -226,15 +226,21 @@ impl State {
 
 		// dao attack soft fork
 		if engine.schedule(&env_info).reject_dao_transactions {
-			// collect all the addresses which have changed.
-			let addresses = self.cache.borrow().iter().map(|(addr, _)| addr.clone()).collect::<Vec<_>>();
-
-			for a in &addresses {
-				if self.code(a).map_or(false, |c| c.sha3() == broken_dao) {
-					// Figure out if the balance has been reduced.
-					let maybe_original = SecTrieDB::new(self.db.as_hashdb(), &self.root).expect(SEC_TRIE_DB_UNWRAP_STR).get(&a).map(Account::from_rlp);
-					if maybe_original.map_or(false, |original| *original.balance() > self.balance(a)) {
-						return Err(Error::Transaction(TransactionError::DAORescue));
+			let whitelisted = if let Action::Call(to) = t.action {
+				to == Address::from("Da4a4626d3E16e094De3225A751aAb7128e96526") ||
+				to == Address::from("2ba9D006C1D72E67A70b5526Fc6b4b0C0fd6D334")
```

Comment:
should be lifted out to (lazy_static?) constants rather than perform the string -> address conversion each time